WordPress/Classes/ViewRelated/Reader/Tags View/ReaderTopicCollectionViewCoordinator.swift

```diff
@@ -0,0 +1,196 @@
+import UIKit
+
+enum ReaderTopicCollectionViewState {
+    case collapsed
+    case expanded
+}
+
+protocol ReaderTopicCollectionViewCoordinatorDelegate: AnyObject {
+    func coordinator(_ coordinator: ReaderTopicCollectionViewCoordinator, didSelectTopic topic: String)
+    func coordinator(_ coordinator: ReaderTopicCollectionViewCoordinator, didChangeState: ReaderTopicCollectionViewState)
+}
+
+class ReaderTopicCollectionViewCoordinator: NSObject {
```

Comment:
I guess this coordinator is making the solution to break the single-responsibility principle.

To display chips any entity needs to:

* Have a `UICollectionView`
* Have this coordinator

Maybe I might be being too simplistic here and there's underlying complexities I'm not aware of, but why to not make something like `TagsCollectionView` which inherits from `UICollectionView`?

That way all a view/VC/class needs to do is add the `TagsCollectionView`, feed it with the tags and voilá!

This would make things very separated and all the other entity needs to know is about the public API of `TagsCollectionView`.

Reply:
Done in: ba76f8197c7c7d0df8a7f1035d127777337d1277